src/libsync/abstractnetworkjob.cpp

```diff
@@ -164,6 +164,27 @@ void AbstractNetworkJob::slotFinished()
         qCWarning(lcNetworkJob) << "SslHandshakeFailedError: " << errorString() << " : can be caused by a webserver wanting SSL client certificates";
     }
 
+    // Qt doesn't yet transparently resend HTTP2 requests, do so here
+    QByteArray verb = requestVerb(*reply());
+    if (_reply->error() == QNetworkReply::ContentReSendError
+        && _reply->attribute(QNetworkRequest::HTTP2WasUsedAttribute).toBool()
+        && (!_requestBody || _requestBody->isSequential())
+        && !verb.isEmpty()) {
```

Comment:
Should we also not increment and check _redirectCount (or another variable) in case there is a problem with Qt or the server that causes QNetworkReply::ContentReSendError to be send over and over again.

Reply:
Yes, definitely!